tests/integrationv2/test_serialization.py

```diff
@@ -0,0 +1,93 @@
+import pytest
+import copy
+import os
+import random
+import string
+
+from configuration import available_ports
+from common import ProviderOptions, Protocols
+from fixtures import managed_process  # lgtm [py/unused-import]
+from providers import Provider, S2N
+from utils import invalid_test_parameters, get_parameter_name, to_bytes
+
+SERVER_STATE_FILE = 'server_state'
+CLIENT_STATE_FILE = 'client_state'
+
+SERVER_DATA = f"Some random data from the server:" + "".join(
+    random.choice(string.ascii_uppercase + string.digits) for _ in range(10)
+)
```

Comment:
Time for this to be a utility / common method? You've done it in two tests now :)